File: lib/src/entity/ble_location.dart
import 'ble_base.dart';

class BleLocation extends BleBase<BleLocation> {
  int mTime = 0; // 距离当地2000/1/1 00:00:00的秒数
  int mActivityMode = 0;
  int mAltitude = 0; // m
  double mLongitude = 0;
  double mLatitude = 0;

  BleLocation();

  @override
  String toString() {
    return "BleLocation(mTime=$mTime, mActivityMode=$mActivityMode, mAltitude=$mAltitude, mLongitude=$mLongitude, mLatitude=$mLatitude)";
  }

  factory BleLocation.fromJson(Map map) {
    BleLocation location = BleLocation();
    location.mTime = map["mTime"];
    location.mActivityMode = map["mActivityMode"];
    location.mAltitude = map["mAltitude"];
    location.mLongitude = map["mLongitude"];
    location.mLatitude = map["mLatitude"];
    return location;
  }

  static List<BleLocation> jsonToList(Map map) {
    List<BleLocation> list = [];
    List tmp = map["list"];
    for (map in tmp) {
      list.add(BleLocation.fromJson(map));
    }
    return list;
  }
}

File: ios/Classes/iOSSmaBle/entity/BleLocation.swift
//
// Created by Best Mafen on 2019/9/26.
// Copyright (c) 2019 szabh. All rights reserved.
//

import Foundation

open class BleLocation: BleReadable {
    static let ITEM_LENGTH = 16

    public var mTime: Int = 0 // 距离当地2000/1/1 00:00:00的秒数
    public var mActivityMode: Int = 0
    public var mAltitude: Int = 0 // m
    public var mLongitude: Float = 0.0
    public var mLatitude: Float = 0.0

    override func decode() {
        super.decode()
        mTime = Int(readInt32())
        mActivityMode = Int(readUInt8())
        skip(8)
        mAltitude = Int(readInt16())
        mLongitude = readFloat()
        mLatitude = readFloat()
    }

    open override var description: String {
        "BleLocation(mTime: \(mTime), mActivityMode: \(mActivityMode), mAltitude: \(mAltitude), mLongitude: \(mLongitude), mLatitude: \(mLatitude))"
    }
    
    public func toDictionary()->[String:Any]{
        let dic : [String : Any] = ["mTime":mTime,
                                    "mActivityMode":mActivityMode,
                                    "mAltitude":mAltitude,
                                    "mLongitude":mLongitude,
                                    "mLatitude":mLatitude]
        return dic
    }
    
    public func dictionaryToObjct(_ dic:[String:Any]) ->BleLocation{

        let newModel = BleLocation()
        if dic.keys.count<1{
            return newModel
        }
        newModel.mTime = dic["mTime"] as? Int ?? 0
        newModel.mActivityMode = dic["mActivityMode"] as? Int ?? 0
        newModel.mAltitude = dic["mAltitude"] as? Int ?? 0
        newModel.mLongitude = dic["mLongitude"] as? Float ?? 0
        newModel.mLatitude = dic["mLatitude"] as? Float ?? 0
        return newModel
    }
}
